sim/src/main/cc/firesim/firesim_top.cc

```diff
@@ -570,7 +570,7 @@ void firesim_top_t::run() {
 
     while (!simulation_complete() && !has_timed_out()) {
         run_scheduled_tasks();
-        step(get_largest_stepsize(), false);
+        take_steps(get_largest_stepsize(), false);
```

Comment:
So it looks like `step` was deleted and thus you replaced it with this. Can you elaborate a bit on what `step` did? I think my confusion is around what `t` means/meant (I know it moved to the `peek_poke` class).

Reply:
t was essentially a cycle count, vis-a-vis steps taken by peek poke. If you used non-blocking peeks / pokes (rare) this would be pretty non-sensical. 

What `step` and it's underlying implementation `take_steps` does is write (indirectly) to the peek-poke widget some number of credits. When the credits are exhausted it stops producing tokens and the simulator eventually stalls. So "steps" here is in terms of cycles in whatever clock domain the peek-poke widget is in. 
We should probably change how time-control is handled to make more sense in a multi-clock, peek-poke-less simulator (in my dissertation branch the hub model will not advance past a certain time thats been plumbed out to MMIO), and rename this.